src/Microsoft.Graph.Core/Authentication/MsalAuthenticationProviderOption.cs

```diff
@@ -0,0 +1,22 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using Microsoft.Identity.Client;
+    using System.Security;
+
+    /// <summary>
+    /// Options class used to configure the authentication providers.
+    /// </summary>
+    internal class MsalAuthenticationProviderOption : IAuthenticationProviderOption
```

Comment:
I'd rename it ScopedAuthenticationProviderOptions since there's nothing related specifically to MSAL here.